docs/design/libraries/ComInterfaceGenerator/ParameterOwnershipAndLifetimes.md

```diff
@@ -0,0 +1,155 @@
+# Ownership and Lifetimes of parameters
```

Comment:
The content here is good. Perhaps there is a more effective way to organize/represent it. Have you considered the pivot being the type and then have examples (both C# and C/C++) for each. Below is a sketch, but not the type not the indirection is the primary pivot. Then below it are the ways to pass the type with an example in C# and the same signature in C/C++/IDL. This let's users easily determine the type they are using and then find an example on how it is being used.

```
## Primitives

### By Value

### By Reference

C# - void Method(ref int a);
COM - HRESULT Method(int* a);

## Array

### By Value - Blittable

C# - void Method(int[] a);
COM - HRESULT Method(int* a);

```